data_structure.py

```diff
@@ -927,3 +927,37 @@ def real_process(node):
         return real_process
 
     return decorator
+
+
+# EDGE CACHE settings : used to accellerate the (linear) edge list generation
+_edgeCache = {}
+_edgeCache["main"] = []  # e.g. [[0, 1], [1, 2], ... , [N-1, N]] (extended as needed)
+
+
+def update_edge_cache(n):
+    '''
```

Comment:
these look like backticks?

Reply:
I checked. they are not. I can replace them with normal quotes if that’s  more desirable.

Reply:
no matter, docstring syntax is unfortunately mixed in data_structure.py. so asking you to use double quotes would be almost arbitrary.